### PR TITLE
Merge 1.3.2 to1.4.0

### PR DESCRIPTION
**Pull request by peacewong**
### What is the purpose of the change

Merge 1.3.2 to1.4.0, Need to use merge to merge

### CODE REVIEW

**Comment by peacewong** [comment]
ping @jackxu2011 @casionone @rarexixi  Need to use merge to merge

File: linkis-dist/package/db/linkis_ddl.sql [hunk 1]
```diff
@@ -42,6 +42,9 @@ CREATE TABLE `linkis_ps_configuration_config_key`(
   `is_advanced` tinyint(1) DEFAULT NULL COMMENT 'Whether it is an advanced parameter. If set to 1(true), parameters would be displayed only when user choose to do so',
   `level` tinyint(1) DEFAULT NULL COMMENT 'Basis for displaying sorting in the front-end. Higher the level is, higher the rank the parameter gets',
   `treeName` varchar(20) DEFAULT NULL COMMENT 'Reserved field, representing the subdirectory of engineType',
+  `en_description` varchar(200) DEFAULT NULL COMMENT 'english description',
+  `en_name` varchar(100) DEFAULT NULL COMMENT 'english name',
+  `en_treeName` varchar(100) DEFAULT NULL COMMENT 'english treeName',
```
Comment (GuoPhilipse) [inline linkis-dist/package/db/linkis_ddl.sql:47]: do we need to add alter command of `linkis_ps_configuration_config_key` in our upgrade sql file?

File: linkis-dist/package/db/linkis_ddl.sql [hunk 2]
```diff
@@ -907,6 +911,9 @@ CREATE TABLE `linkis_ps_dm_datasource_type`
     `classifier`  varchar(32) COLLATE utf8_bin NOT NULL,
     `icon`        varchar(255) COLLATE utf8_bin DEFAULT NULL,
     `layers`      int(3)                       NOT NULL,
+    `description_en` varchar(255) DEFAULT NULL COMMENT 'english description',
```
Comment (GuoPhilipse) [inline linkis-dist/package/db/linkis_ddl.sql:914]: do we need to add alter command of `linkis_ps_dm_datasource_type ` in our upgrade sql file?

File: linkis-dist/package/db/linkis_ddl.sql (hunk 1 above)
Comment (peacewong) [reply linkis-dist/package/db/linkis_ddl.sql:47]: Yes, this is already in package/db/upgrade/1.3.2_schema/mysql/linkis_ddl.sql

File: linkis-dist/package/db/linkis_ddl.sql (hunk 2 above)
Comment (peacewong) [reply linkis-dist/package/db/linkis_ddl.sql:914]: Yes, this is already in package/db/upgrade/1.3.2_schema/mysql/linkis_ddl.sql

File: .github/workflows/build-backend.yml [hunk 3]
```diff
@@ -20,24 +20,72 @@ name: Build Backend
...
+      uses: actions/setup-java@v3
       with:
         distribution: 'temurin'
         java-version: 8
-    - name: Build backend by maven
+        cache: maven
+    - if: ${{matrix.spark == '2.4' && matrix.hadoop == '3.3'}}
+      name: build spark ${{matrix.spark}} and hadoop ${{matrix.hadoop}}
+      run:
+        ./mvnw clean package -Pspark-${{matrix.spark}}-hadoop-${{matrix.hadoop}} -Phadoop-${{matrix.hadoop}} -Dmaven.test.skip=true
```
[the remark this answers is not in the document]
Comment (peacewong) [reply .github/workflows/build-backend.yml:47]: Thank you, I will modify it

**Comment by jackxu2011** [comment]
is the 1.4.0 should add review required?

**Comment by peacewong** [comment]
> is the 1.4.0 should add review required?

Yes, we need to modify the strategy of the master, you can do this

**Pull request merged**